src/components/update-service/update-service.controller.ts

```diff
@@ -0,0 +1,355 @@
+      onShow: this.showConfig
+    };
+    this.reviewStep = {
+      label: 'Results',
+      id: 'results',
+      view: 'update-service/update-service-results.html',
+      hidden: false,
+      allowed: false,
+      valid: true,
+      prevEnabled: false,
+      allowClickNav: false,
+      onShow: this.showResults
+    };
+
+    this.ctrl.steps = [this.planStep, this.configStep, this.reviewStep];
+
+    this.ctrl.orderedPlans = _.orderBy(this.ctrl.servicePlans, ['spec.externalMetadata.displayName', 'metadata.name']);
+
+    this.configChanged = false;
+    this.ctrl.displayName = this.$filter('serviceInstanceDisplayName')(this.ctrl.serviceInstance, this.ctrl.serviceClass);
+    this.ctrl.serviceName = _.get(this.ctrl.serviceInstance, 'metadata.name');
+
+    if (this.planStep.hidden) {
+      this.ctrl.hideBackButton = true;
+    }
+
+    this.context = {
+      namespace: _.get(this.ctrl.serviceInstance, 'metadata.namespace')
+    };
+
+    this.origParameterData = angular.copy(_.get(this.ctrl.serviceInstance, 'spec.parameters', {}));
+
+    var promises: any = [];
+
+    _.each(_.get(this.ctrl.serviceInstance, 'spec.parametersFrom'), (parametersSource) => {
+      var secretName = _.get(parametersSource, 'secretKeyRef.name');
+      var secret = _.find(this.secrets, function(nextSecret: any) {
+        return _.get(nextSecret, 'metadata.name') === secretName;
+      });
+
+      if (secret) {
+        this.addParametersFromSecret(secret, parametersSource);
+      } else {
+        promises.push(this.DataService.get("secrets", secretName, this.context).then((secret: any) => {
+          this.addParametersFromSecret(secret, parametersSource);
+          this.secrets.push(secret);
+        }));
+      }
+    });
+
+    this.$q.all(promises).then(() => {
+      this.originalPlan = _.find(this.ctrl.orderedPlans, (orderedPlan: any) => {
+        return _.get(orderedPlan, 'metadata.name') === _.get(this.ctrl.serviceInstance, 'spec.clusterServicePlanRef.name');
+      });
+
+      this.selectPlan(this.originalPlan);
+    });
+  };
+
+  public clearValidityWatcher = () => {
+    if (this.validityWatcher) {
+      this.validityWatcher();
+      this.validityWatcher = undefined;
+    }
+    this.ctrl.reviewStep.allowed = false;
+  };
+
+  public showPlan = () => {
+    this.clearValidityWatcher();
+    this.ctrl.configPageShown = false;
+
+    var updateProperties: any = _.get(this.ctrl.parameterSchema, 'properties');
+    this.configStep.hidden = _.size(updateProperties) < 1;
+
+    if (this.configStep.hidden) {
+      this.ctrl.nextTitle = 'Update';
+    } else {
+      this.ctrl.nextTitle = "Next >";
+    }
+    this.planStep.valid =  (this.ctrl.selectedPlan !== this.originalPlan) || !this.configStep.hidden;
+
+    this.validityWatcher = this.$scope.$watch("$ctrl.parameterSchema", () => {
+      var updateProperties: any = _.get(this.ctrl.parameterSchema, 'properties');
+      this.configStep.hidden = _.size(updateProperties) < 1;
```

Comment:
Might pull these two lines into a helper since it's repeated from above.

Reply:
Actually all of this can just be done whenever we update the parameterSchema at that time (no need to watch). Thanks.